pkg/client/typed/discovery/discovery_client.go

```diff
@@ -110,7 +110,7 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList
 	if err == nil {
 		apiGroup = apiVersionsToAPIGroup(v)
 	}
-	if err != nil && !errors.IsNotFound(err) && !errors.IsForbidden(err) {
```

Comment:
I actually don't know why this was this way in the first place (no tests broke when I changed it). Need to look into the implications a little more. Anyone have some insight?